freqtrade/data/converter/orderflow.py

```diff
@@ -0,0 +1,295 @@
+"""
+Functions to convert orderflow data from public_trades
+"""
+
+import logging
+import time
+import typing
+from collections import OrderedDict
+from datetime import datetime
+from typing import Dict, Tuple
+
+import numpy as np
+import pandas as pd
+
+from freqtrade.constants import DEFAULT_ORDERFLOW_COLUMNS
+from freqtrade.exceptions import DependencyException
+
+
+logger = logging.getLogger(__name__)
+
+# Global cache dictionary
+cached_grouped_trades_per_pair: Dict[str, OrderedDict[Tuple[datetime, datetime], pd.DataFrame]] = {}
```

Comment:
using a global constant like this will almost certainly cause huge problems in at least 3 cases i can think of right now
* tests run at random - but this cache will "outlive" them - so you might encounter test pollution - with random tests depending on their sequence / the parallel worker they're assigned on. While you can reset this after every test - forget one and you'll spend weeks trying to trace down a random test failure (yes, this happened to me ...).
* "/reload" will NOT clear this - so you can end up with a memory leak - as a "/reload" could change "everything" - including the database and the stake-currency ... (as well as pairs, obviously).
* Backtests with multiple strategies will not have this cleared. Not entirely sure if that's a problem / usecase we can even support due to memory constraints - but it's going to be problematic for sure if people are able to run it.

instead, this cache must live in an object ... most likely (based on the fact from where `populate_dataframe_with_trades` is called) within the strategy class.
That will also make the function have no side-effects - which makes it a lot simpler to test